### ISSUE TITLE
add documentation for cmake install

### ISSUE THREAD

**Issue by MCWertGaming** [jupyter-xeus/cpp-terminal#198]
We should add documentation on installing the library using cmake:
- [ ] install on linux as shared library
- [ ] use cpp-terminal as shared library on linux
- [ ] install cpp-terminal as shared library on windows
- [ ] using shared libraries on windows
  - [ ] place into shared include dir?
  - [ ] place next to the dll
  - [ ] static linking
  - [ ] other methods? Maybe visual studio stuff
- [ ] MacOS specific things?
- [ ] cross platform approach for managing cpp-terminal using cmake
  - [ ] git submodules
  - [ ] cmake managed
  - [ ] mamba?
  - [ ] other ways / package managers?

**Comment by ShalokShalom** [comment]
Currently, there is all the documentation about CMake missing: 

https://github.com/jupyter-xeus/cpp-terminal/wiki/Adding-CPP-Terminal-to-your-ptoject

In the end, the docs recommend two approaches, which are both not covered. 
The only option remains the highly complicated one, that is .. not recommended.

P.S: Also, you mention in this documentation, that the project consisted of two header files in 2021, and the README says it was just one. 

Thanks a lot

**Comment by Giddy-Up224** [comment]
Yeah! Frustrating how the documentation [promises](https://github.com/jupyter-xeus/cpp-terminal/wiki/Adding-CPP-Terminal-to-your-ptoject#using-cmake-only) (like AI) that it has something for me that it really doesn't have:

> To get you started more quickly, we have prepared a simple cmake function to get you started immediately:

**Comment by flagarde** [comment]
@Giddy-Up224 Hi, I can understand your frustration but please note that this project is developped by many people in a free-time basis... A lot of API/ABI changed...

For CMake there is not documentation but for people aware of the CMake philosophie it should not be so difficult to install/use... it works fine with fetchcontent for example.

**Comment by flagarde** [comment]
@Giddy-Up224  There is a basic test for a find_package() https://github.com/jupyter-xeus/cpp-terminal/blob/master/tests%2Ftest-standalone%2FCMakeLists.txt

**Comment by flagarde** [comment]
@Giddy-Up224 I personally don't agree with the readme.. IMHO I would prefer a find_package for already installed library or a FetchContent for compiling as part of your project.... I agree thga the readme is not up-to-date... feel free to add a PR

for an example (I use the library for my own project see https://github.com/yaodaq/YAODAQ/blob/main/cmake%2Fcpp-terminal.cmake

this project try to follow CMake standards

**Comment by Giddy-Up224** [comment]
> [@Giddy-Up224](https://github.com/Giddy-Up224) Hi, I can understand your frustration but please note that this project is developped by many people in a free-time basis... A lot of API/ABI changed...
> 
> For CMake there is not documentation but for people aware of the CMake philosophie it should not be so difficult to install/use... it works fine with fetchcontent for example.

@flagarde thanks for the reply and sorry if I was too sarcastic. I didn't realize that `FetchContent` works with this library. That should make it fairly easy. I just had no clue where to start. I'm a junior dev and learning CMake, so things like this can feel overwhelming. If I end up using this repo *(and I likely will, sooner or later)* I'll try to create a PR with my findings to help the next searching soul.

🫰All the best...

**Comment by flagarde** [comment]
@Giddy-Up224 no problem. We tried to adapt good practice for CMake so it should work with not so many problem.  If you find some problem please feel free to create an issue.

main reasons why I have not touched much the readme is :

1) I still have a lot of idea tonimplement in the library but lack of time

2) bad frenchglish 😛

**Comment by Giddy-Up224** [comment]
> * I still have a lot of idea tonimplement in the library but lack of time
> * bad frenchglish 😛

Maybe you could create a PR and have a native English speaker approve and edit it? BTW, your English isn't that bad. It's better than some native English speakers using poor grammar. (or bad language)

All the best...
